src/components/Common/UI/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,145 @@
+import type { InputHTMLAttributes, FocusEvent } from 'react'
+import { useRef } from 'react'
+import {
+  DatePicker as AriaDatePicker,
+  Button,
+  Calendar,
+  CalendarCell,
+  CalendarGrid,
+  DateInput,
+  DateSegment,
+  Dialog,
+  Group,
+  Heading,
+  Popover,
+  type DateValue,
+} from 'react-aria-components'
+import { useTranslation } from 'react-i18next'
+import classNames from 'classnames'
+import { CalendarDate } from '@internationalized/date'
+import type { SharedFieldLayoutProps } from '../FieldLayout'
+import { FieldLayout } from '../FieldLayout'
+import { useFieldIds } from '../hooks/useFieldIds'
+import styles from './DatePicker.module.scss'
+import { useTheme } from '@/contexts/ThemeProvider'
+import CaretDown from '@/assets/icons/caret-down.svg?react'
+import CaretRight from '@/assets/icons/caret-right.svg?react'
+import CaretLeft from '@/assets/icons/caret-left.svg?react'
+
+// Utility functions to convert between Date and DateValue
+function dateToDateValue(date: Date | null | undefined): DateValue | null {
+  if (!date) return null
+
+  return new CalendarDate(
+    date.getFullYear(),
+    date.getMonth() + 1, // JavaScript months are 0-indexed
+    date.getDate(),
+  )
+}
+
+function dateValueToDate(dateValue: DateValue | null): Date | null {
+  if (!dateValue) return null
+
+  return new Date(
+    dateValue.year,
+    dateValue.month - 1, // DateValue months are 1-indexed
+    dateValue.day,
+  )
+}
+
+export interface DatePickerProps
```

Comment:
Nit: could we consider adding an `inputRef` to the DatePicker props so that the input focus for react hook form on failed submission works?

Reply:
Good call out, will be in next iteration